### DISCUSSION TITLE
I have no idea why $GO_REVISION is coming blank

### DISCUSSION THREAD

**Discussion by iambenkay** [gocd/gocd#10765] [Q&A]
Hey guys, I started using GOCD a while ago,
I already added a material to my pipeline and my jobs run perfectly without the $GO_REVISION variable.
I recently added this step in my job that needs to use $GO_REVISION but for some reason it was not working. I added a step before it to echo the value of $GO_REVISION and it was an empty string as seen below:
<img width="528" alt="image" src="https://user-images.githubusercontent.com/36256097/188271714-23fad2e0-a80b-47ca-a8a8-54bfe6bc5838.png">

Has anyone faced this before? How can I resolve?

**Comment by iambenkay** [comment]
```sh
/bin/sh -c "printenv | grep GO_"
```
I ran the above and discovered that GO_REVISION was not being set instead GO_REVISION_${MATERIAL_NAME} was being set.
According to the docs this should happen when there are multiple materials in the pipeline but I have just one material

**Reply by chadlwilson** [reply]
The docs possibly need clarifying; from the code it looks to me like if the material has a friendly name **or** a custom checkout folder set, it will always be included in the env var.

Does your material have either of these set?

**Reply by iambenkay** [reply]
Yes it does. It has a material name

**Reply by iambenkay** [reply]
Can you point me to what part of the code is responsible?

**Reply by chadlwilson** [reply]
Sure: https://github.com/gocd/gocd/blob/5c7c6c226dc2caa1fe729f5142536634912e795a/domain/src/main/java/com/thoughtworks/go/config/materials/ScmMaterial.java#L200-L222

**Reply by iambenkay** [reply]
So is this a candidate for a PR or do we update docs?
It could cause confusion if nothing is done.

**Reply by chadlwilson** [reply]
A PR would be welcome to clarify in the docs - I don't see the behaviour being changed, as it'd be a breaking change and has been this way likely for a long time. Docs are at https://github.com/gocd/docs.go.cd/blob/master/content/faq/dev_use_current_revision_in_build.md
